[Streams] Add EBT browser events

mykolaharmash opened the pull request:
Partially resolves https://github.com/elastic/streams-program/issues/186

This adds EBT browser events for these actions in the UI:
* Processing saved
* Stream's retention config changed
* Child stream created
* Schema changed (field is mapped or unmapped)
* Failure store config changed
* Significant event suggestions generated
* Significant events created

Some events also include additional metadata. See their schema definitions for details.

## How to test

Go through the mentioned UI actions and make sure you see corresponding events sent via `kibana-browser` endpoint.

mykolaharmash committed:
[Streams] Add EBT browser events

jbudz reviewed (approved):
`packages/kbn-optimizer/limits.yml` LGTM

mykolaharmash committed:
Add stream type to events

mykolaharmash committed:
Merge remote-tracking branch 'upstream/main' into streams-telemetry-events

mykolaharmash committed:
Generalize the failure store event

mykolaharmash committed:
Merge remote-tracking branch 'upstream/main' into streams-telemetry-events

mykolaharmash committed:
Revert to switch for lifecycle detection

mykolaharmash committed:
Increse bundle limit

mykolaharmash committed:
Merge remote-tracking branch 'upstream/main' into streams-telemetry-events

mykolaharmash committed:
Add stream_type to processing_saved event

mykolaharmash committed:
Fix import after merge conflict

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/339685)
* Commit: 699ca2bddade93f043f2d90c3cf46d1d925dfeb5

### Failed CI Steps
* [Build Kibana Distribution](https://buildkite.com/elastic/kibana-pull-request/builds/339685#019951cb-495b-4bb7-b757-0ea807e536d5)

### History
* :broken_heart: [Build #339659](https://buildkite.com/elastic/kibana-pull-request/builds/339659) failed 7c7de685e345bfd404b2021a1c97b0fbb700637c
* :broken_heart: [Build #339456](https://buildkite.com/elastic/kibana-pull-request/builds/339456) failed 3d08acbc3cd52ad043c1a63cfe344ae40331fe45
* :broken_heart: [Build #339437](https://buildkite.com/elastic/kibana-pull-request/builds/339437) failed fe70808e764e226af9a830b4273e9e59f15bfb31
* :broken_heart: [Build #339302](https://buildkite.com/elastic/kibana-pull-request/builds/339302) failed cb33b230faeafc49746e0369383aae35442580dd
* :green_heart: [Build #338749](https://buildkite.com/elastic/kibana-pull-request/builds/338749) succeeded 99b2f0ef09a51240262c65dc99ae4d3b3c15fa03
* :broken_heart: [Build #338646](https://buildkite.com/elastic/kibana-pull-request/builds/338646) failed 7420fa2b5ad0b56c9bab55039af331415efe6705

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339685","number":339685,"commit":"699ca2bddade93f043f2d90c3cf46d1d925dfeb5"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339659","number":339659,"commit":"7c7de685e345bfd404b2021a1c97b0fbb700637c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339456","number":339456,"commit":"3d08acbc3cd52ad043c1a63cfe344ae40331fe45"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339437","number":339437,"commit":"fe70808e764e226af9a830b4273e9e59f15bfb31"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339302","number":339302,"commit":"cb33b230faeafc49746e0369383aae35442580dd"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/338749","number":338749,"commit":"99b2f0ef09a51240262c65dc99ae4d3b3c15fa03"}],"number":339685}
buildkite-pr-comment-->

mykolaharmash committed:
Add wired streams status change event

mykolaharmash committed:
Merge remote-tracking branch 'upstream/main' into streams-telemetry-events

mykolaharmash commented:
* Added another event for tracking status of wired streams as @flash1293 suggested
* Added a helper function to detect stream type from its definition as @dgieselaar suggested

mykolaharmash committed:
Fix typing issue

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/339929)
* Commit: d67b4b31b5ca9b5167543aaf47e56ea1093e87a4
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-234777-d67b4b31b5ca`

### History
* :broken_heart: [Build #339898](https://buildkite.com/elastic/kibana-pull-request/builds/339898) failed 45d0a5d00aea641d9056b69e46a8c7bbf2d259a8
* :broken_heart: [Build #339685](https://buildkite.com/elastic/kibana-pull-request/builds/339685) failed 699ca2bddade93f043f2d90c3cf46d1d925dfeb5
* :broken_heart: [Build #339659](https://buildkite.com/elastic/kibana-pull-request/builds/339659) failed 7c7de685e345bfd404b2021a1c97b0fbb700637c
* :broken_heart: [Build #339456](https://buildkite.com/elastic/kibana-pull-request/builds/339456) failed 3d08acbc3cd52ad043c1a63cfe344ae40331fe45
* :broken_heart: [Build #339437](https://buildkite.com/elastic/kibana-pull-request/builds/339437) failed fe70808e764e226af9a830b4273e9e59f15bfb31
* :broken_heart: [Build #339302](https://buildkite.com/elastic/kibana-pull-request/builds/339302) failed cb33b230faeafc49746e0369383aae35442580dd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339929","number":339929,"commit":"d67b4b31b5ca9b5167543aaf47e56ea1093e87a4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339898","number":339898,"commit":"45d0a5d00aea641d9056b69e46a8c7bbf2d259a8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339685","number":339685,"commit":"699ca2bddade93f043f2d90c3cf46d1d925dfeb5"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339659","number":339659,"commit":"7c7de685e345bfd404b2021a1c97b0fbb700637c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339456","number":339456,"commit":"3d08acbc3cd52ad043c1a63cfe344ae40331fe45"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339437","number":339437,"commit":"fe70808e764e226af9a830b4273e9e59f15bfb31"}],"number":339929}
buildkite-pr-comment-->

flash1293 reviewed (approved):
Retested and LGTM

mykolaharmash committed:
Merge remote-tracking branch 'upstream/main' into streams-telemetry-events

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/340180)
* Commit: 53e54e5843cfc63d7ef87cba23d2aa34227df48f

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/340180#019956ae-5c5f-4b54-a845-a1367206784d)

### History
* :green_heart: [Build #339929](https://buildkite.com/elastic/kibana-pull-request/builds/339929) succeeded d67b4b31b5ca9b5167543aaf47e56ea1093e87a4
* :broken_heart: [Build #339898](https://buildkite.com/elastic/kibana-pull-request/builds/339898) failed 45d0a5d00aea641d9056b69e46a8c7bbf2d259a8
* :broken_heart: [Build #339685](https://buildkite.com/elastic/kibana-pull-request/builds/339685) failed 699ca2bddade93f043f2d90c3cf46d1d925dfeb5
* :broken_heart: [Build #339659](https://buildkite.com/elastic/kibana-pull-request/builds/339659) failed 7c7de685e345bfd404b2021a1c97b0fbb700637c
* :broken_heart: [Build #339456](https://buildkite.com/elastic/kibana-pull-request/builds/339456) failed 3d08acbc3cd52ad043c1a63cfe344ae40331fe45

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340180","number":340180,"commit":"53e54e5843cfc63d7ef87cba23d2aa34227df48f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339929","number":339929,"commit":"d67b4b31b5ca9b5167543aaf47e56ea1093e87a4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339898","number":339898,"commit":"45d0a5d00aea641d9056b69e46a8c7bbf2d259a8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339685","number":339685,"commit":"699ca2bddade93f043f2d90c3cf46d1d925dfeb5"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339659","number":339659,"commit":"7c7de685e345bfd404b2021a1c97b0fbb700637c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339456","number":339456,"commit":"3d08acbc3cd52ad043c1a63cfe344ae40331fe45"}],"number":340180}
buildkite-pr-comment-->

mykolaharmash committed:
fixup! Merge remote-tracking branch 'upstream/main' into streams-telemetry-events

mykolaharmash committed:
Merge remote-tracking branch 'upstream/main' into streams-telemetry-events
